presto-jdbc/src/main/java/io/prestosql/jdbc/ConnectionProperties.java

```diff
@@ -309,23 +309,55 @@ public ExtraCredentials()
         // E.g., `jdbc:presto://example.net:8080/?extraCredentials=abc:xyz;foo:bar` will create credentials `abc=xyz` and `foo=bar`
         public static Map<String, String> parseExtraCredentials(String extraCredentialString)
         {
-            return Splitter.on(';').splitToList(extraCredentialString).stream()
-                    .map(ExtraCredentials::parseSingleCredential)
+            return new MapPropertyParser("credential", "name", "value").parse(extraCredentialString);
```

Comment:
map's name is `extraCredentials`, not `credential`:

```suggestion
            return new MapPropertyParser("extraCredentials", "name", "value").parse(extraCredentialString);
```

Reply:
The way these parameters are used in the messages requires you to look into actual impl to know what values can be provided here. I'd just drop keyName, valueName parameters and use "key" and "value" always.

Also, there seems to be little benefit from this being an object, so:

```suggestion
            return MapPropertyParser.parse("extraCredentials", extraCredentialString);
```